### DISCUSSION TITLE
Access token과 Refresh token에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by BaekYuri** [wanted-9th-3team/frontend-study#60] [프론트 기술 질문 공간]
아는만큼 설명해주세요.

**Comment by michoball** [comment]
oAuth 를 이용할 때 사용하게 됩니다. 
oAuth는 제 3자 서비스에 응용 프로그램이나 서비스의 인증을 위한 접근 권한을 위임하는 방법을 제공하여 웹이나 모바일 앱에 안전하게 사용자 인증 및 권한 부여를 위한 개방형 표준 프로토콜입니다. 

사용자에게 서비스를 제공하는 기업은 미리 기업의 서비스이용에 필요한 제 3의 서비스 플랫폼에 미리 기업 인증을 받아둡니다. 

이 인증 과정으로 얻은 비밀 키를 가지고 새로운 사용자가 올 때마다 3의 기업에 새로운 사용자에게 인증을 받아 3의 기업의 정보를 사용할 수 있도록 해달라고 요청합니다. 

이때 3의 기업이 사용자에게 계정 정보를 요구하여 계정정보를 얻고 기업에 사용자에대한 인증 토큰을 발급합니다.  이 인증 토큰이 access_token입니다. 

기업은 이 인증 토큰을 이용해서 사용자가 3의 기업에서 사용중인 정보 ( 예:  스케줄, 달력, 메모, 로그인 등등)를 얻을 수 있습니다. 

이 인증 토큰은 보안을 위해 유효시간이 정해져 있습니다. 이 시간이 지난 후에는 이 인증 토큰을 사용하지 못하는데요. 이때 앞선 과정을 반복하지 않고 보다 간편하게 다시 인증 토큰을 발급받기 위해서 있는 것이 refresh token 입니다. 

인증 토큰이 만료되었다는 응답을 받으면 이 토큰을 보내서 새로운 인증 토큰을 발급받아 계속해서 3의 기업의 정보를 제공 받을 수 있습니다.

**Comment by WooGie911** [comment]
JWT(JSON WEB TOKEN)에서 서비스의 인증을 위해 사용됩니다.
Header에 Access token과 Refresh token을 담아 사용자를 인증하며, 로그인시 Access token과 Refresh token 두가지를 모두 발급합니다.
그리고 Access token을 사용해 사용자를 구분하고, Access token이 만료시 Refresh token을 사용하여 Access token을 재발급합니다.

**Comment by BaekYuri** [comment]
액세스 토큰과 리프레시 토큰은 OAuth 인증에서 사용되는 두 가지 유형의 토큰입니다. 액세스 토큰은 보호된 리소스에 대한 액세스 권한을 부여하는 데 사용되는데, 제 3자에게 탈취당한다면 보안에 취약해질 수 있습니다. 그러므로 액세스 토큰의  유효기간을 짧게 하고 리프레시 토큰을 이용해 액세스 토큰을 재발급합니다.
액세스 토큰은 클라이언트가 보호된 리소스에 액세스할 때 서버에 전송됩니다. 리프레시 토큰은 클라이언트가 액세스 토큰을 재발급할 때 서버에 전송됩니다.
